graph.py

```diff
@@ -0,0 +1,85 @@
+class Graph(object):
+    """
+    Create the graph class and initialize an empty dictionary
+    """
+    def __init__(self):
+        self.g = {}
+
+    def nodes(self):
+        """
+        returns a list of nodes
+        """
+        return self.g.keys()
+
+    def edges(self):
+        """
+        returns a list of tuples the corresponds to the edges
+        """
+        result = []
+        for key, values in self.g.iteritems():
+            for value in values:
+                if value:
+                    result.append((key, value))
+        return result
+
+    def add_node(self, n):
+        """
+        Add a node to the Graph
+        """
+        self.g[n] = []
```

Comment:
what happens if `n` is already in the graph and has edges from it?  Write a test to expose this problem and then fix it.